runtime-sdk/src/context.rs

```diff
@@ -177,6 +178,9 @@ pub struct TxContext<'a, 'b> {
 
     /// Per-context values.
     values: &'b mut BTreeMap<&'static str, Box<dyn Any>>,
+
+    /// Per-transaction values.
+    tx_values: BTreeMap<&'static str, Box<dyn Any>>,
```

Comment:
should this also be borrowed as `&'b mut`?

Reply:
this one is owned, with the lifecycle matching the context itself